lib/msf/core/payload/apk.rb

```diff
@@ -95,18 +95,43 @@ def fix_manifest(tempdir)
     File.open("#{tempdir}/original/AndroidManifest.xml", "wb") {|file| file.puts original_manifest.to_xml }
   end
 
-  def backdoor_apk(apkfile, raw_payload)
+  def find_orig_cert_dname(orig_apkfile)
+    unzip_output = run_cmd("unzip -l #{orig_apkfile}")
+    orig_cert_file = unzip_output.match(/[\w\/-]+\.RSA/)[0]
+    bin_cert_data, err, s = Open3.capture3("unzip -p #{orig_apkfile} #{orig_cert_file}")
+    unless s.success?
+      raise RuntimeError, "Failed to find original cert data."
+    end
+    keytool_output, err, s = Open3.capture3("keytool -printcert", :stdin_data=>bin_cert_data, :binmode=>true)
```

Comment:
I think we should use the -jarfile argument of keytool, e.g:
keytool_output = run_cmd("keytool -printcert -jarfile #{orig_apkfile}")
then we can remove the dependency on unzip.